### PR TITLE
Add support for creating Delta tables with deletion vectors

### LINKED ISSUE #22104: Add support for creating tables with deletion vector in Delta Lake connector
Follow-up of https://github.com/trinodb/trino/pull/22102

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Fixes #22104

## Release notes

```markdown
# Delta Lake
* Add support for creating tables with deletion vectors. ({issue}`22104`)
```

### CODE REVIEW

**Review by anusudarsan — APPROVED** [review]

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTableProperties.java [hunk 1]
```diff
@@ -129,4 +134,9 @@ public static ColumnMappingMode getColumnMappingMode(Map<String, Object> tablePr
     {
         return ColumnMappingMode.valueOf(tableProperties.get(COLUMN_MAPPING_MODE_PROPERTY).toString().toUpperCase(ENGLISH));
     }
+
+    public static boolean getDeletionVectorsEnabled(Map<String, Object> tableProperties)
+    {
+        return (boolean) tableProperties.getOrDefault(DELETION_VECTORS_ENABLED_PROPERTY, false);
```
Comment (raunaqmorarka) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTableProperties.java:140]: Can we pick up the default from a catalog property ? That way admins can enable it for all new tables without requiring users to set it in every table definition.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java [hunk 2]
```diff
@@ -996,6 +999,61 @@ public void testAllowColumnDefaults()
         assertQueryFails("INSERT INTO allow_column_defaults (a) VALUES (2)", "\\QUnsupported writer features: [allowColumnDefaults]");
     }
```
Comment (raunaqmorarka) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java:1001]: Is altering table to add/remove this property supported ?
Can we add a test for that ?

Comment (raunaqmorarka) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java:1001]: Can we also have a product test that table with deletion vector created by Trino is readable by DB/Spark ?

**Comment by ebyhr** [comment]
@raunaqmorarka Addressed comments.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTableProperties.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTableProperties.java:140]: Added `delta.deletion-vectors-enabled` config property.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java (hunk 2 above)
Comment (ebyhr) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java:1002]: It's unsupported for now. Added a negative test.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeConfig.java [hunk 3]
```diff
@@ -84,6 +84,7 @@ public class DeltaLakeConfig
     private boolean registerTableProcedureEnabled;
     private boolean projectionPushdownEnabled = true;
     private boolean queryPartitionFilterRequired;
+    private boolean deletionVectorsEnabled;
```
Comment (raunaqmorarka) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeConfig.java:87]: What's the default for this in delta/databricks ?

**Review by raunaqmorarka — APPROVED** [review]

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeConfig.java (hunk 3 above)
Comment (ebyhr) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeConfig.java:87]: OSS Delta Lake disables it by default. Databricks depends on the version (it's disabled at version < 14.x).

Comment (raunaqmorarka) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeConfig.java:87]: If latest Databricks enables it by default, we should work towards enabling this by default as well in future

**Pull request merged**